File: runtime/runtime_delegate.cc
// Copyright 2015 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "flutter/runtime/runtime_delegate.h"

namespace blink {

RuntimeDelegate::~RuntimeDelegate() {}

void RuntimeDelegate::DidCreateMainIsolate(Dart_Isolate isolate) {}

void RuntimeDelegate::DidCreateSecondaryIsolate(Dart_Isolate isolate) {}

void RuntimeDelegate::DidShutdownMainIsolate() {}

}  // namespace blink

File: engine/src/flutter/lib/web_ui/lib/src/engine/compositor/runtime_delegate.dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// @dart = 2.6
part of engine;

abstract class RuntimeDelegate {
  String get defaultRouteName;
  void scheduleFrame({bool regenerateLayerTree = true});
  void render(LayerTree layerTree);
  void handlePlatformMessage(PlatformMessage message);
  FontCollection getFontCollection();
}
